org.eclipse.lsp4e/src/org/eclipse/lsp4e/outline/CNFOutlinePage.java

```diff
@@ -73,11 +72,7 @@ public CNFOutlinePage(LanguageServer languageServer, @Nullable ITextEditor textE
 		preferences = InstanceScope.INSTANCE.getNode(LanguageServerPlugin.PLUGIN_ID);
 		preferences.addPreferenceChangeListener(this);
 		this.textEditor = textEditor;
-		if (textEditor != null) {
-			this.textEditorViewer = ((ITextViewer) textEditor.getAdapter(ITextOperationTarget.class));
-		} else {
-			this.textEditorViewer = null;
-		}
+		this.textEditorViewer = textEditor == null ? null : textEditor.getAdapter(ITextViewer.class);
```

Comment:
I think this will now return null in many cases. The reason to use ITextOperationTarget is a hack: the getAdapter() method won't return a viewer if queried for the viewer but null; querying for the ITextOperationTarget usually returns the viewer.
One possibility would be first to extract some method `getTextViewer(ITextEditor editor)`. In this method, we can first try adapting to ITextViewer (ideally with `Adapters.getAdapter(editor, ITextViewer.cclass)` instead of the direct call to `editor.getAdapter()`), then if this return null try adapting to ITextOperationTarget, and verify result is instanceof ITextViewer before returning it, or null.

Reply:
done